Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -465,11 +528,18 @@ struct VersionAssignmentSet<C: PackageContainer>: Equatable, Sequence {
             // A package can be excluded if there are no constraints on the
             // package (it has not been requested by any other package in the
             // assignment).
-            return constraints[container.identifier] == .any
+            return constraints[container.identifier] == .versionSet(.any)
 
         case .version(let version):
             // A version is valid if it is contained in the constraints.
-            return constraints[container.identifier].contains(version)
+            if case .versionSet(let versionSet) = constraints[container.identifier] {
+                return versionSet.contains(version)
+            }
+            return false
```

Comment:
Based on the `unversioned` wins principle, I think "mathematically" it is valid to merge any version with an existing unversioned constraint. Don't think it actually changes any behavior, though.